storagedriver.go

```diff
@@ -72,6 +73,18 @@ func NewMemoryStorage(backendStorageName string) (*memory.InMemoryStorage, error
 			*argDbTable,
 			*argDbName,
 		)
+	case "redis":
+		var hostname string
+		hostname, err = os.Hostname()
```

Comment:
Hostname() won't return what you expect if cAdvisor is running inside a container. You probably want to accept an IP or hostname as a flag.